subsys/caf/modules/sensor_sampler.c

```diff
@@ -134,35 +152,46 @@ static bool is_active(const struct sensor_config *sc, struct sensor_data *sd,
 	return is_active;
 }
 
-static bool can_sensor_sleep(const struct sensor_config *sc,
-			     struct sensor_data *sd,
-			     const float *curr)
+static void reset_sensor_trigger_counter(const struct sensor_config *sc,
+					 struct sensor_data *sd)
 {
-	size_t data_cnt = get_sensor_data_cnt(sc);
-	bool sleep = true;
+	unsigned int timeout = sc->trigger ? sc->trigger->activation.timeout_ms : 0;
```

Comment:
Similarly as in `reset_sensor_trigger_counter` function. I don't think we should check if `sc->trigger` is there each time. Just `timeout = sc->trigger->activation.timeout_ms`;